File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const dayArr = []
    for(let i=0; i<progresses.length; i+=1){
        dayArr.push(Math.ceil((100-progresses[i])/speeds[i]))
    }
    console.log(dayArr)
    let n = dayArr.shift();
    let c = 1;
    const result = []
    while(dayArr.length){
        const d = dayArr.shift();
        if(n<d){
            result.push(c)
            n=d;
            c=1;
            continue;
        }
        c+=1;
    }
    
    result.push(c)
    return result
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    queue = deque()
    for i in range(len(progresses)):
        queue.append(math.ceil((100-progresses[i])/speeds[i]))
        
    start = None
    count = 0
    result = []
    while(queue):
        if(start==None):
            start = queue.popleft()
            count += 1
            continue
        if(start>=queue[0]):
            queue.popleft()
            count += 1
        else:
            result.append(count)
            start=None
            count=0
    result.append(count)
    return result
    